DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageFunctionGenerator.java

```diff
@@ -558,6 +560,118 @@ public static void main(String args[]) {
             append(buf, castFromObjFormatter, BinaryExpr.Operator.PLUS, c, Object.class);
         }
 
+        // Special casts for PyObject to primitive
+        buf.append("        public static int intPyCast(Object a) {\n");
+        buf.append("            if (a != null && !(a instanceof PyObject)) {\n");
+        buf.append("                throw new IllegalArgumentException(\"Provided value is not a PyObject\");\n");
+        buf.append("            }\n");
+        buf.append("            PyObject o = (PyObject) a;\n");
+        buf.append("            if (o == null || o.isNone()) {\n");
+        buf.append("                return QueryConstants.NULL_INT;\n");
+        buf.append("            }\n");
+        buf.append("            return o.getIntValue();\n");
+        buf.append("        }\n\n");
+
+        buf.append("        public static double doublePyCast(Object a) {\n");
+        buf.append("            if (a != null && !(a instanceof PyObject)) {\n");
+        buf.append("                throw new IllegalArgumentException(\"Provided value is not a PyObject\");\n");
+        buf.append("            }\n");
+        buf.append("            PyObject o = (PyObject) a;\n");
+        buf.append("            if (o == null || o.isNone()) {\n");
+        buf.append("                return QueryConstants.NULL_DOUBLE;\n");
+        buf.append("            }\n");
+        buf.append("            return o.getDoubleValue();\n");
+        buf.append("        }\n\n");
+
+        buf.append("        public static long longPyCast(Object a) {\n");
+        buf.append("            if (a != null && !(a instanceof PyObject)) {\n");
+        buf.append("                throw new IllegalArgumentException(\"Provided value is not a PyObject\");\n");
+        buf.append("            }\n");
+        buf.append("            PyObject o = (PyObject) a;\n");
+        buf.append("            if (o == null || o.isNone()) {\n");
+        buf.append("                return QueryConstants.NULL_LONG;\n");
+        buf.append("            }\n");
+        buf.append("            return o.getLongValue();\n");
+        buf.append("        }\n\n");
+
+        buf.append("        public static float floatPyCast(Object a) {\n");
+        buf.append("            if (a != null && !(a instanceof PyObject)) {\n");
+        buf.append("                throw new IllegalArgumentException(\"Provided value is not a PyObject\");\n");
+        buf.append("            }\n");
+        buf.append("            PyObject o = (PyObject) a;\n");
+        buf.append("            if (o == null || o.isNone()) {\n");
+        buf.append("                return QueryConstants.NULL_FLOAT;\n");
+        buf.append("            }\n");
+        buf.append("            return floatCast(o.getDoubleValue());\n");
```

Comment:
```suggestion
        buf.append("            return (float) o.getDoubleValue();\n");
```
The change is needed to avoid the spurious handling of special null values.